JDP190901-5

pawban opened the pull request:
**UserController - wstępna implementacja endpointów**

Utworzyłem klasy UserDto oraz LoginDto w pakiecie
com.kodilla.ecomercee.dto. Klasy odpowiadają specyfikacjom zawartym
w plikach users.spec.json oraz login.spec.json.
Kontroler oznaczyłem odpowiednimi adnotacjami i utworzyłem metody
odpowiadające enpointom wymaganym w module 25.5.
W kontrolerze utowrzyłem również listę z przykładowymi użytkownikami,
na których tymczasowo pracuje kontroler - do czasu pełnej implementacji.

pawban committed:
UserController - wstępna implementacja endpointów

Utworzyłem klasy UserDto oraz LoginDto w pakiecie
com.kodilla.ecomercee.dto. Klasy odpowiadają specyfikacjom zawartym
w plikach users.spec.json oraz login.spec.json.
Kontroler oznaczyłem odpowiednimi adnotacjami i utworzyłem metody
odpowiadające enpointom wymaganym w module 25.5.
W kontrolerze utowrzyłem również listę z przykładowymi użytkownikami,
na których tymczasowo pracuje kontroler - do czasu pełnej implementacji.

pawban committed:
UserController - wstępna implementacja endpointów

Cała logika (wszystkie testy poprawności przesłanych danych, generowanie
przykładowych danych) została przeniesiona do osobnej klasy UserService
w pakiecie service.

pawban committed:
Merge branch 'master' into JDP190901-5

pawban committed:
UserController - wstępna implementacja endpointów (poprawiona)

Kontroler został uproszczony - teraz zwraca po prostu przykładowe dane.
W klasach UserDto oraz LoginDto zamieniłem konstruktory i gettery na
stosowne andotacje Lomboka.

pawban commented:
Kontroler został uproszczony - teraz zwraca po prostu przykładowe dane.
W klasach UserDto oraz LoginDto zamieniłem konstruktory i gettery na
stosowne andotacje Lomboka.

Arelias commented on src/main/java/com/kodilla/ecommercee/dto/UserDto.java:
```diff
@@ -0,0 +1,17 @@
+
+    private Long id;
+    private String username;
+    private Integer status;
```
Status uzytkownika (tylko zgaduje) zapewne jest online lub offline, wtedy mozna byloby uzyc booleana zamiast Integera, ale jesli projekt zaklada, ze te pole ma byc Integer, to ok. Innych uwag do tego kodu nie posiadam.

mryndak replied:
Do przechowywania statusów polecam użyć enuma

pawban replied:
Też o tym myślałem, ale jesteśmy związani specyfikacją - zwracany JSON musi być identyczny, jak wyspecyfikowany przykład - w przykładzie pole status ma wartość liczbową.

mryndak replied:
Nie jest to problemem: https://stackoverflow.com/a/37833901 
https://stackoverflow.com/a/56285491
ale pozostawiam wg uznania :)

pawban committed:
Merge branch 'master' into JDP190901-5

mryndak commented on src/main/java/com/kodilla/ecommercee/dto/LoginDto.java:
```diff
@@ -0,0 +1,15 @@
+@Getter
+public class LoginDto {
+
+    private Boolean loggedIn;
```
Jeżeli wartość ta będzie przyjmować tylko true / false, to lepiej użyć boolean. Boolean - może być null

pawban replied:
Fakt, że wartość wrappera może przybierać wartość `null` był właśnie powodem zastosowania typu `Boolean` - żeby mieć pewność, że wartość tego pola została ustawiona, a nie zainicjowana wartością domyślną. Choć przyznaję, że po zastanowieniu wydaje mi się, że akurat w tym miejscu niekoniecznie jest to istotne.

mryndak commented on build.gradle:
```diff
@@ -29,7 +29,7 @@ dependencies {
     implementation('mysql:mysql-connector-java:8.0.17')
     implementation('org.projectlombok:lombok:1.18.10')
-    annotationProcessor 'org.projectlombok:lombok'
+    annotationProcessor('org.projectlombok:lombok')
```
Nie potrzebna modyfikacja

pawban replied:
Wiem, ale ten brak nawiasów bił mnie po oczach.
Niemniej - wycofałem.

pawban committed:
UserController - wstępna implementacja (zmiana typu pola w LoginDto)

Typ pola loggedIn w klasie LoginDto został zmieniony z typu
opakowującego Boolean na typ prosty boolean.

pawban committed:
Merge remote-tracking branch 'origin/JDP190901-5' into JDP190901-5

pawban committed:
UserController - wstępna implementacja (wycofanie zbędnej zmiany)

Zbędna zmiana w pliku build.gradle dotycząca wpisu
annotationProcessor została wycofana.

Arelias committed:
Merge branch 'master' into JDP190901-5
